src/main/java/ch/njol/skript/effects/EffStopSound.java

```diff
@@ -90,45 +88,38 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 			category = (Expression<SoundCategory>) exprs[1];
 			players = (Expression<Player>) exprs[2];
 		}
-		
 		return true;
 	}
 
 	@Override
 	protected void execute(Event event) {
 		// All sounds pattern wants explicitly defined master category
 		SoundCategory category = this.category == null ? null : this.category.getOptionalSingle(event)
-			.orElse(allSounds ? null : SoundCategory.MASTER);
+				.orElse(allSounds ? null : SoundCategory.MASTER);
+
 		Player[] targets = players.getArray(event);
-		
 		if (allSounds) {
 			if (category == null) {
-				for (Player player : targets) {
+				for (Player player : targets)
 					player.stopAllSounds();
-				}
 			} else {
-				for (Player player : targets) {
+				for (Player player : targets)
 					player.stopSound(category);
-				}
 			}
-		} else {
+		} else if (sounds != null) {
```

Comment:
I think an assertion would fit better here. `allSounds` being true should imply `sounds != null`, right?

Reply:
Yes, but an assertion won't stop execution for a normal user, only in the test environment.